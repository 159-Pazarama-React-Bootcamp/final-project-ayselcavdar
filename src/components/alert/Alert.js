import styles from './Alert.module.css';
import PropTypes from 'prop-types';

const Alert = ({ info, error, title, body }) => {
  return (
    <div
      className={`${styles["alert-container"]} ${
        info ? styles["alert-info"] : ""
      } ${error ? styles["alert-error"] : ""}`}
    >
      {title && (
        <div className={styles["title-container"]}>
          <h3 className={styles["title-text"]}>{title}</h3>
        </div>
      )}
      <div className={styles["body-container"]}>
        <p className={styles["body-text"]}>{body}</p>
      </div>
    </div>
  );
};

export default Alert;

Alert.propTypes = {
  info: PropTypes.bool,
  error: PropTypes.bool,
  title: PropTypes.string,
  body: PropTypes.string.isRequired,
};
