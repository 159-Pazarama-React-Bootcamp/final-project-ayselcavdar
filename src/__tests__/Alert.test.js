import render from 'react-test-renderer';
import Alert from '../components/alert/Alert';

describe('Alert comp change className to props', () => {
  it('if prop type info, classname should has alert-info', () => {
    const wrapper = shallow(<Alert info body="hello" />);
    expect(wrapper.find('.alert-info').getElements().length).toEqual(1);
  });

  it("if required prop exist, it should be in p tag as a text", () => {
    const wrapper = shallow(<Alert info body="hello" />);
    expect(wrapper.find("p").text()).toEqual("hello");
  });

  it('if prop type error', () => {
    const wrapper = shallow(<Alert error body="error" />);
    expect(wrapper.find('.alert-error').getElements().length).toEqual(1);

  });

  it("if required prop exist, it should be in p tag as a text", () => {
    const wrapper = shallow(<Alert error body="error" />);
    expect(wrapper.find("p").text()).toEqual("error");
  });

  it('Alert render check', () => {
    const comp = render.create(<Alert info body="informational"/>);
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
