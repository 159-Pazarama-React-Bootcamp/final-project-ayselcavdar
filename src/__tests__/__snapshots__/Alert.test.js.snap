// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert comp change className to props Alert render check 1`] = `
<div
  className="alert-container alert-info "
>
  <div
    className="body-container"
  >
    <p
      className="body-text"
    >
      informational
    </p>
  </div>
</div>
`;

exports[`Alert render check 1`] = `
<div
  className="alert-container alert-info "
>
  <div
    className="body-container"
  >
    <p
      className="body-text"
    >
      informational
    </p>
  </div>
</div>
`;
